### PR TITLE
DiagnosticScope related changes

### PR DESCRIPTION
**Pull request by ravithanneeru**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by ravithanneeru** [comment]
@cochi2 Please review from your side as well.

**Comment by cochi2** [comment]
@ravithanneeru could you please provide more context and description of the changes? Thanks.

**Comment by ravithanneeru** [comment]
> @ravithanneeru could you please provide more context and description of the changes? Thanks.

We are trying to add live tests for all APIs and found that if we don't add diagnostic scope, the API failing with following error. So, added for the missing APIs including download recording APIs.

Expected diagnostic scope not created CallingServerClient.GetParticipant 
    created 2 scopes [CallingServerClient.GetParticipantAsync, Azure.Core.Http.Request] 
    You may have forgotten to add clientDiagnostics.CreateScope(...), set your operationId to CallingServerClient.GetParticipant in GetParticipantAsync or applied the Azure.Core.ForwardsClientCallsAttribute to GetParticipantAsync.

File: sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs [hunk 1]
```diff
@@ -536,13 +536,25 @@ public virtual Response ResumeRecording(string recordingId, CancellationToken ca
         public virtual async Task<Response<Stream>> DownloadStreamingAsync(
             Uri sourceEndpoint,
             HttpRange range = default,
-            CancellationToken cancellationToken = default) =>
-            await _contentDownloader.DownloadStreamingInternal(
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallingServerClient)}.{nameof(DownloadStreaming)}");
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:541]: Are we ignoring the `Async` suffix of the methods? How will we know which method is the failing one if two methods share the same scope naming?

Comment (ravithanneeru) [reply sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:541]: I have the same question on how DiagnosticScope works, @zihzhan-msft Can you please throw some info on this?

Comment (RyanZzzz) [reply sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:541]: https://teams.microsoft.com/l/message/19:97e6cbbb188d4f8e9e7ce0f189f362ae@thread.skype/1638815675144?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1638815675144&teamName=Azure%20SDK&channelName=Azure%20Core&createdTime=1638815675144

Create a thread in Azure Core channel for the scope name between Sync and Async methid.

Comment (zihzhan-msft) [reply sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:541]: https://teams.microsoft.com/l/message/19:97e6cbbb188d4f8e9e7ce0f189f362ae@thread.skype/1638815675144?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1638815675144&teamName=Azure%20SDK&channelName=Azure%20Core&createdTime=1638815675144

Create a thread in Azure Core channel for the scope name between Sync and Async methid.

Comment (zihzhan-msft) [reply sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:541]: @ravithanneeru Per the answer in the thread, we should remove `Async` postfix for all the api operations, can you help on that?

Comment (ravithanneeru) [reply sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:541]: @zihzhan-msft we never used Async in any of the CreateScope calls.

**Comment by ravithanneeru** [comment]
Merge changes showing up as part of changes.

**Pull request closed without merging**